python/ray/data/dataset.py

```diff
@@ -2317,6 +2319,10 @@ def take_batch(
         except StopIteration:
             raise ValueError("The dataset is empty.")
         self._synchronize_progress_bar()
+
+        # Save the computed stats and blocks to the original dataset.
+        self._plan._snapshot_stats = limited_ds._plan.stats()
+        self._plan._snapshot_blocks = limited_ds._plan._snapshot_blocks
```

Comment:
why we have to do it now?

Reply:
if we don't save the stats + blocks, which are computed from the dataset w/ limit applied, back to the original dataset, we will need to recompute them. there are also a number of tests which use `ds.take()`, then examine the `ds.stats()` to check for stages, etc.

Reply:
hmm but that sounds incorrect, right? Support we do

```
result = ds.take_batch(batch_size=10) # expect to return a batch with 10 rows
result = ds.take_batch(batch_size=100) # expect to return a batch with 100 rows, but only returns 10 rows here?
```